src/Microsoft.DotNet.CodeAnalysis/DiagnosticIds.cs

```diff
@@ -12,5 +12,9 @@ internal enum DiagnosticIds
         BCL0010, // AppContextDefaultNotInitializedToTrueDiagnostic
         BCL0011, // AppContextDefaultUsedUnexpectedIfStatement
         BCL0012, // DefaultValueDefinedOutsideIfCondition
+
+        BCL0100, // PinvokeCallCheck
```

Comment:
Since this is not a flags enum we can probably leave about 5 numbers per analyzer (in case new warnings are added to existing analyzers). I would make these be BCL0015 and BCL0020.